stablehlo/dialect/StablehloOps.td

```diff
@@ -1259,6 +1259,7 @@ def StableHLO_AllToAllOp : StableHLO_Op<"all_to_all",
 def StableHLO_ReduceOp: StableHLO_ShapedInterfaceOp<"reduce", [
       RecursiveSideEffects,
       SameVariadicOperandSize,
+      InferTensorTypeWithReify,
```

Comment:
Did you mean `InferTensorType`?

Reply:
There exists a `ReduceOp::reifyReturnTypeShapes()`. So `InferTensorTypeWithReify` seems more proper here as it is fit for op which has both: "inferReturnTypeComponents" and "reifyReturnTypeShapes".